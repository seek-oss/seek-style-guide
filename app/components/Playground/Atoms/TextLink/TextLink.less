@import (reference) "~seek-style-guide/theme";

.link {
  .link();
  display: inline-block;
}

.chevron {
  margin-left: 4px;
}

.chevronSvg {
  height: 14px;
  width: 14px;
}

.subheading {
  .subheadingTextResponsive();
  padding-bottom: @grid-row-height;
}

.heading {
  .headingTextResponsive();
  padding-bottom: @grid-row-height;
}

.hero {
  .heroTextResponsive();
  padding-bottom: @grid-row-height;
}

.emphasis {
  font-weight: @sk-bold;
}

.touchable {
  .touchableText(@standard-type-scale);
}
