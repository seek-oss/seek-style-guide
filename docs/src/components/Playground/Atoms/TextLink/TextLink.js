import styles from './TextLink.less';
import React from 'react';
import PropTypes from 'prop-types';

import ChevronIcon from 'seek-style-guide/react/ChevronIcon/ChevronIcon';
import classnames from 'classnames';

const renderChevron = chevron => {
  if (!chevron) {
    return null;
  }

  return (
    <ChevronIcon
      className={styles.chevron}
      direction={chevron}
      svgClassName={styles.chevronSvg}
    />
  );
};

export default function TextLink({ component: Root, className, subheading, heading, hero, children, chevron, ...restProps }) {
  const allProps = {
    ...restProps,
    className: classnames(styles.link, {
      [className]: className,
      [styles.subheading]: subheading,
      [styles.heading]: heading,
      [styles.hero]: hero,
      [styles.touchable]: !subheading && !heading && !hero
    })
  };

  return (
    <Root {...allProps}>
      {children}
      {renderChevron(chevron)}
    </Root>
  );
}

TextLink.displayName = 'TextLink';

TextLink.propTypes = {
  component: PropTypes.any,
  className: PropTypes.string,
  children: PropTypes.node,
  chevron: PropTypes.oneOf(['up', 'down', 'right', 'left']),
  subheading: PropTypes.bool,
  heading: PropTypes.bool,
  hero: PropTypes.bool
};

TextLink.defaultProps = {
  component: 'a'
};
