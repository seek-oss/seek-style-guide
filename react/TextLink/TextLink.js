import styles from './TextLink.less';

import React from 'react';
import PropTypes from 'prop-types';

import ChevronIcon from '../ChevronIcon/ChevronIcon';
import classnames from 'classnames';

export const chevronDirections = ['up', 'down', 'right', 'left'];

const renderChevron = (chevron, chevronProps, isLeftChevron) => {
  if (!chevron) {
    return null;
  }

  return (
    <ChevronIcon
      {...chevronProps}
      className={
        isLeftChevron ? styles.chevronLeftSide : styles.chevronRightSide
      }
      direction={chevron}
    />
  );
};

export default function TextLink({
  component: Root,
  className,
  children,
  chevron,
  chevronProps,
  ...restProps
}) {
  const allProps = {
    ...restProps,
    className: classnames(styles.link, className)
  };

  const isLeftChevron = chevron === 'left';

  return (
    <Root {...allProps}>
      {isLeftChevron && renderChevron(chevron, chevronProps, isLeftChevron)}
      {children}
      {!isLeftChevron && renderChevron(chevron, chevronProps, isLeftChevron)}
    </Root>
  );
}

TextLink.displayName = 'TextLink';

TextLink.propTypes = {
  component: PropTypes.any,
  className: PropTypes.string,
  children: PropTypes.node,
  chevron: PropTypes.oneOf(chevronDirections),
  chevronProps: PropTypes.object
};

TextLink.defaultProps = {
  component: 'a',
  chevronProps: { size: 'standard' }
};
