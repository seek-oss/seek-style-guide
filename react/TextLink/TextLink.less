@import (reference) "~seek-style-guide/theme";

.link {
  .link();
}

@chevron-spacing: 4px;

.chevronRightSide {
  margin-left: @chevron-spacing;
}

.chevronLeftSide {
  margin-right: @chevron-spacing;
}
