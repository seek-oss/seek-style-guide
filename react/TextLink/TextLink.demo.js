import React from 'react';
import PropTypes from 'prop-types';
import TextLink from './TextLink';
import Text from '../Text/Text';

const TextLinkContainer = ({ component: DemoComponent, componentProps }) => (
  <Text>
    <DemoComponent {...componentProps} />
  </Text>
);
TextLinkContainer.propTypes = {
  component: PropTypes.any,
  componentProps: PropTypes.object.isRequired
};

export default {
  route: '/text-link',
  title: 'Text Link',
  category: 'Typography',
  component: TextLink,
  container: TextLinkContainer,
  initialProps: {
    href: '#',
    children: 'Text link'
  },
  options: [
    {
      label: 'Chevron',
      type: 'radio',
      states: [
        {
          label: 'No chevron',
          transformProps: props => props
        },
        {
          label: 'Up',
          transformProps: props => ({
            ...props,
            chevron: 'up'
          })
        },
        {
          label: 'Down',
          transformProps: props => ({
            ...props,
            chevron: 'down'
          })
        },
        {
          label: 'Right',
          transformProps: props => ({
            ...props,
            chevron: 'right'
          })
        },
        {
          label: 'Left',
          transformProps: props => ({
            ...props,
            chevron: 'left'
          })
        }
      ]
    }
  ]
};
