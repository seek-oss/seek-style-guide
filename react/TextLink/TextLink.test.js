import { shallow } from 'enzyme';
import React from 'react';
import TextLink, { chevronDirections } from './TextLink';

describe('TextLink', () => {
  it('should render with defaults', () => {
    expect(shallow(<TextLink />)).toMatchSnapshot();
  });

  it('should render with className', () => {
    expect(shallow(<TextLink className="foo" />)).toMatchSnapshot();
  });

  it('should render with children', () => {
    expect(shallow(<TextLink>Google</TextLink>)).toMatchSnapshot();
  });

  chevronDirections.forEach(direction => {
    it(`should render with ${direction} chevron`, () => {
      expect(
        shallow(<TextLink chevron={direction}>Google</TextLink>)
      ).toMatchSnapshot();
    });
  });

  it('should render with chevron props', () => {
    expect(
      shallow(
        <TextLink chevron="up" chevronProps={{ size: 'hero' }}>
          Google
        </TextLink>
      )
    ).toMatchSnapshot();
  });
});
