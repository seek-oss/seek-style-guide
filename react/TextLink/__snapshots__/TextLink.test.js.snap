// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink should render with chevron props 1`] = `
<a
  className="TextLink__link"
>
  Google
  <ChevronIcon
    className="TextLink__chevronRightSide"
    direction="up"
    size="hero"
  />
</a>
`;

exports[`TextLink should render with children 1`] = `
<a
  className="TextLink__link"
>
  Google
</a>
`;

exports[`TextLink should render with className 1`] = `
<a
  className="TextLink__link foo"
/>
`;

exports[`TextLink should render with defaults 1`] = `
<a
  className="TextLink__link"
/>
`;

exports[`TextLink should render with down chevron 1`] = `
<a
  className="TextLink__link"
>
  Google
  <ChevronIcon
    className="TextLink__chevronRightSide"
    direction="down"
    size="standard"
  />
</a>
`;

exports[`TextLink should render with left chevron 1`] = `
<a
  className="TextLink__link"
>
  <ChevronIcon
    className="TextLink__chevronLeftSide"
    direction="left"
    size="standard"
  />
  Google
</a>
`;

exports[`TextLink should render with right chevron 1`] = `
<a
  className="TextLink__link"
>
  Google
  <ChevronIcon
    className="TextLink__chevronRightSide"
    direction="right"
    size="standard"
  />
</a>
`;

exports[`TextLink should render with up chevron 1`] = `
<a
  className="TextLink__link"
>
  Google
  <ChevronIcon
    className="TextLink__chevronRightSide"
    direction="up"
    size="standard"
  />
</a>
`;
